### DISCUSSION TITLE
I'm getting kinda git error

### DISCUSSION THREAD

**Discussion by iaminamcom** [Homebrew/discussions#1170] [Linux] [answered]
On running `brew install package` getting following error:
```
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

What to do now? I followed [this](https://stackoverflow.com/questions/39836190/homebrew-install-failed-during-git-fetch-origin-masterrefs-remotes-origin-mas) guide but I think on linux we have brew directories in /home/homebrew/ so I tried deleting things in there but even becomes complex. Can you please help me?  
It worked fine after installation but after I did git global config it started acting up.

Thank you!

**Comment by gromgit** [comment]
What's the output of `brew config` and `brew doctor`?

**Reply by iaminamcom** [reply]
It's 
For `brew config`
```bash
HOMEBREW_VERSION: 3.0.10-55-g4f7bd24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4f7bd240e1fc120f2ae16f264b165d2a39803097
Last commit: 32 hours ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 2
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dual-core 64-bit penryn
Clang: N/A
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 5.4.0-58-generic x86_64 GNU/Linux
OS: Linux Mint 20.1 (ulyssa)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: 2.7.0
glibc: N/A
gcc@5: N/A
xorg: N/A
```

and for `brew doctor`
```bash
Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout 
```

**Reply by iaminamcom** [reply]
After running `git -C $(brew --repo homebrew/core) checkout` getting:
fatal: You are on a branch yet to be born

**Reply by gromgit** [reply]
In your case, the Homebrew core tap isn't complete:
```
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
```
Do the following:
```
rm -fr $(brew --repo homebrew/core)
brew tap homebrew/core
```
If you don't see any errors, run `brew config` again and look at the lines quoted above. In particular, `Core tap last commit:` should be a time interval, not `never`.

**Reply by iaminamcom** [reply]
Ran `brew tap homebrew/core`
and getting this
```
Updating Homebrew...
==> Tapping homebrew/core
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1421, done.
remote: Counting objects: 100% (1421/1421), done.
remote: Compressing objects: 100% (537/537), done.
fatal: the remote end hung up unexpectedly4.54 MiB | 16.00 KiB/s
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/linuxbrew-core /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core --origin=origin` exited with 128.
```

Can I uninstall and reinstall it ?

**Reply by gromgit** [reply]
It looks more like you're having trouble completing the core tap cloning, so reinstalling probably wouldn't help.

What's the output of `cat $(brew --repo homebrew/core)/.git/config`?

**Reply by iaminamcom** [reply]
Getting
```
cat: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/.git/config: No such file or directory
```

**Reply by gromgit** [reply]
OK, try `brew tap homebrew/core` again.

**Comment by iaminamcom** [comment, marked as answer]
Sorry issue was solved by solving git issue. Git timeout and compression was acting up

**Reply by gromgit** [reply]
> Git timeout and compression was acting up

Can you share details about how you fixed your issue? Git is a fundamental Homebrew infrastructure component, so the more details about what needs to be fixed from an end-user perspective, the more likely that a fix can be devised to improve things for everyone, especially new users.

**Reply by iaminamcom** [reply]
Yes sure.
The issue is related to the git as I changed the config. I freshly installed Linux mint and brew was my first installation it worked perfect but when I added config info I saw error: 
```
Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

and it's not solved.

Things became problematic when you told me to run `rm -fr $(brew --repo homebrew/core)` to remove tap core because tap core repo size is hugely big with thousands of commits and  git was acting up as better consult this thread https://stackoverflow.com/questions/21277806/fatal-early-eof-fatal-index-pack-failed. In my particular case I did as in https://stackoverflow.com/a/29355320/15519865 and also I changed internet speed by using phone 4G internet which is crazily faster.

**BTW, your suggestion to remove core tap seems to work but git issue came separately.**

Thanks for the best support
